Bottom: b032160065f760943507e40ad9033fe529236258
Top:    bfd501521ebef6a9f4c571aba813a26b247c6646
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-11-29 15:49:37 +0800

Refresh of tuntap-napi

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 594aa93..4c01e16 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1192,9 +1192,9 @@ static int tun_enqueue(struct tun_file *tfile, struct sk_buff *skb)
 	if (skb_queue_len(list) >= 1)
 		goto drop;
 	__skb_queue_tail(list, skb);
-	napi_schedule(&tfile->napi);
 	spin_unlock(&list->lock);
 
+	napi_schedule(&tfile->napi);
 	return 0;
 drop:
 	spin_unlock(&list->lock);
